File: server/test_score.ts
import { prisma } from './src/db.js';

async function getProduct() {
  const p = await prisma.product.findFirst({
    where: { name: { contains: 'Baby Teethers' } },
    include: { seller: true }
  });
  if (!p) {
    console.log('Not found');
    return;
  }
  console.log(`ID: ${p.id}`);
  console.log(`Name: ${p.name}`);
  console.log(`Price: ${p.price}`);
  console.log(`Brand ID: ${p.brandId}`);
  console.log(`Avg Rating: ${p.averageRating}`);
  console.log(`Reviews Count: ${p.reviewsCount}`);
  console.log(`Popularity: ${p.popularity}`);
  console.log(`Seller New: ${p.seller.isNewSeller}`);
  console.log(`Seller Rating: ${p.seller.rating}`);
}

getProduct().finally(() => process.exit(0));

File: recommendation-system/test_score.py
import sys
sys.path.append(r'd:\OneDrive\Desktop\udrcrafts\recommendation-system')
from app.db.session import SessionLocal
from app.models import Product, UserBehaviour

db = SessionLocal()
p = db.query(Product).filter(Product.name.like('%Baby Teethers%')).first()
if not p:
    print('Product not found')
else:
    print(f'ID: {p.id}')
    print(f'Name: {p.name}')
    print(f'Price: {p.price}')
    print(f'Category ID: {p.categoryId}')
    print(f'Brand ID: {p.brandId}')
    print(f'Average Rating: {getattr(p, "averageRating", 0)}')
    print(f'Reviews Count: {getattr(p, "reviewsCount", 0)}')
    print(f'Popularity: {getattr(p, "popularity", 0)}')
    
    if p.seller:
        print(f'Seller New: {p.seller.isNewSeller}')
        print(f'Seller Rating: {getattr(p.seller, "rating", 0)}')
